src/common.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg(feature = "volta-updates")]
+use std::process::{Command, ExitStatus};
+
+use volta_core::error::ErrorDetails;
+use volta_core::layout::{volta_home, volta_install};
+use volta_fail::{ResultExt, VoltaError};
+
+pub enum Error {
+    Volta(VoltaError),
+    Tool(i32),
+}
+
+pub fn ensure_layout() -> Result<(), Error> {
+    let home = volta_home().map_err(Error::Volta)?;
+
+    if !home.layout_file().exists() {
+        let install = volta_install().map_err(Error::Volta)?;
+        Command::new(install.migrate_executable())
+            .env("VOLTA_LOGLEVEL", format!("{}", log::max_level()))
+            .status()
+            .with_context(|_| ErrorDetails::CouldNotStartMigration)
+            .into_result()?;
+    }
+
+    Ok(())
+}
+
+pub trait IntoResult<T> {
+    fn into_result(self) -> Result<T, Error>;
+}
```

Comment:
Could this just be a `From` impl? (I imagine no, given this exists, but I'm curious why not!)

Reply:
It can't due to orphaning rules, because we're converting from a `Result<T, E>` to a `Result<U, F>`, so both the source and destination are external types.